content/livesync/models/models.textile

```diff
@@ -0,0 +1,278 @@
+---
+title: Models
+meta_description: "A model is a live, observable data model backed by your database that you can use to keep your frontend applications updated in realtime with the latest information from your database."
+product: livesync
+languages:
+  - javascript
+---
+
+A model is a single instance of a live, observable data model backed by your database. A model requires a "sync function":#sync to tell the model how to initialize with the latest data, and a "merge function":#merge to calculate the next version of the model state when change events are received from the backend database.
+
+You can then "subscribe":#subscribe to a model to receive updates in realtime whenever the model state changes.
+
+h2(#create). Create or retrieve a model
+
+To create the model, use the @models.get()@ method on the client. If a model with the given name already exists, it will be returned.
+
+To instantiate a model you must provide a unique name. This identifies the model on the client, and is also the name of the "channel":/channels used to subscribe to state updates from the backend.
+
+```[javascript]
+const modelsClient = modelsClient.models.get({
+  name: 'myPost',
+  channelName: 'post:123',
+  sync,
+  merge,
+});
+```
+
+<aside data-type='note'>
+<p>Restrictions for model names:</p>
+<ul><li>Avoid starting names with @[@ or @:@</li>
+<li>Ensure names aren't empty</li>
+<li>Exclude whitespace and wildcards, such as @*@</li>
+<li>Use the correct case, whether it be uppercase or lowercase</li></ul>
+</aside>
+
+h2(#sync). Sync functions
+
+A sync function tells your model how to initialize with the latest data from the backend and is required when instantiating a model. It can be any function that optionally accepts some parameters and returns a promise with the latest state of your data model along with a "@sequenceID@":#sequence.
+
+
+The following is an example of a sync function. It retrieves the latest model state from your database:
+
+```[javascript]
+async function sync(id: number, page: number) {
+  const result = await fetch(`/api/post/${id}?page=${page}`);
+  return result.json();
+}
+```
+
```

Comment:
I think we need to show how to register the sync function on the model when it is instantiated? e.g.

```
const model = modelsClient.models.get({
  name: 'myPost',
  channelName: 'post:123',
  sync,
  merge,
});
```